### PR TITLE
Add parent-child relationship for groups

### PR DESCRIPTION
**Pull request by sanjay990**

### CODE REVIEW

File: presto-product-tests/src/main/java/com/facebook/presto/tests/ImmutableLdapObjectDefinitions.java [hunk 1]
```diff
@@ -31,6 +31,8 @@
     private static final String americaOrgDistinguishedName = format("ou=America,%s", domain);
     private static final String asiaOrgDistinguishedName = format("ou=Asia,%s", domain);
     private static final String password = "LDAPPass123";
+    private static final String memberOf_relation = "memberOf";
+    private static final String member_relation = "member";
```
Comment (amrutagokhale) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/ImmutableLdapObjectDefinitions.java]: Please capitalize these strings (and all previous static final strings as well). Perhaps you can also drop the "_relation" from the name.

File: presto-product-tests/src/main/java/com/facebook/presto/tests/ImmutableLdapObjectDefinitions.java [hunk 2]
```diff
@@ -66,21 +68,39 @@ private static LdapObjectDefinition buildLdapOrganizationObject(String id, Strin
...
-                        "member", format("uid=%s,%s", userName, userOrganizationName)))
-                .setObjectClasses(Arrays.asList("groupOfNames"))
-                .build();
+        if (childGroupNames.isPresent() && childGroupOrganizationName.isPresent()) {
+            return LdapObjectDefinition.builder(groupName)
+                    .setDistinguishedName(format("cn=%s,%s", groupName, groupOrganizationName))
+                    .setAttributes(ImmutableMap.of(
+                            "cn", groupName,
+                            "member", format("uid=%s,%s", userName, userOrganizationName)))
+                    .setModificationAttributes(getAttributes(childGroupNames.get(), childGroupOrganizationName.get(), member_relation))
```
Comment (amrutagokhale) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/ImmutableLdapObjectDefinitions.java]: For subgroup relationships, we are only adding a "member" relationship but not "memberOf" relationship. Can you make sure that this one way specification is sufficient?

**Comment by amrutagokhale** [comment]
For the commit message, you can say `Add member relationships between groups`.

Looks good % comments.

**Comment by anusudarsan** [comment]
can you rebase again on ldap-tests branch?

**Pull request merged**